Source/SwiftLintFramework/Rules/ExplicitTypeInterfaceRule.swift

```diff
@@ -29,8 +29,11 @@ public struct ExplicitTypeInterfaceRule: ASTRule, OptInRule, ConfigurationProvid
                          dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
 
         guard configuration.allowedKinds.contains(kind),
-            !containsType(dictionary: dictionary),
-            let offset = dictionary.offset else {
+            !dictionary.containsType,
+            !dictionary.containsWeakSelf,
```

Comment:
shouldn't we try to check if there's a capture group, no matter if it's weak or if it's capturing `self`?

Reply:
Yeah, that would make sense. Thanks for pointing it out!

This case is still failing:
```
 "class Foo {\n"                                         +
            "   func bar() {\n"                                     +
            "       var k = 0\n"                                    +
            "       _ = { [weak self, weak k] in\n"                 +
            "       guard let strongSelf = self else { return }\n"  +
            "       }\n"                                            +
            "   }\n"                                                +
            "}",
```

I am not sure how could I find the capture group, my first guess would be using a regex, something like this:
`=\s+{\s*\[((\s)*\w+\s*\w+(,)*)+]\s+in`

Do you have maybe an other idea how to approach finding the capture group?

Reply:
Added a solution for searching capture groups with regex.